### PR TITLE
Candidate response bug fixes and UI redesign

### PR DESCRIPTION
**Pull request by Mustafa-Esoofally**
image

### CODE REVIEW

File: src/app/admin/CandidateResponses.tsx [hunk 1]
```diff
@@ -56,19 +57,40 @@ const fetchCandidates = async (): Promise<CandidateData[]> => {
 export const CandidateResponses = () => {
     const [modalOpen, setModalOpen] = useState(false);
     const [selectedChallenge, setSelectedChallenge] = useState('');
+    const [searchTerm, setSearchTerm] = useState('');
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
```
Comment (noncuro) [inline src/app/admin/CandidateResponses.tsx:61]: Do you need this? Why not just display the error directly?

File: src/app/api/challenge/start/route.ts [hunk 2]
```diff
@@ -1,37 +1,50 @@
...
-}
\ No newline at end of file
+import { NextResponse } from "next/server";
+import { createRedisClient, getChallengeFromCookie } from "@/app/api/utils";
+
+export async function POST() {
+  const redis = createRedisClient();
+
+  try {
+    const challengeStatus = await getChallengeFromCookie(redis);
```
Comment (noncuro) [inline src/app/api/challenge/start/route.ts:8]: Do we have to check if the challenge is already started?

File: src/app/api/challenge/submit/route.ts [hunk 3]
```diff
@@ -1,66 +1,74 @@
...
+      return NextResponse.json(
+        { success: false, message: "Overtime submission not allowed" },
+        { status: 400 },
+      );
+    }
 
-        // Save updated challenge status
-        await redis.set(`challenge:${challengeStatus.emailAddress}`, JSON.stringify(challengeStatus));
+    // Update challenge status
+    challengeStatus.submission = submission;
```
Comment (noncuro) [inline src/app/api/challenge/submit/route.ts:44]: Can we change this to being a list of submissions that include timestamps for each? We can still show just the latest in the frontend

**Review by noncuro — APPROVED** [review]

File: src/app/api/challenge/start/route.ts (hunk 2 above)
Comment (Mustafa-Esoofally) [reply src/app/api/challenge/start/route.ts:8]: We are fetching the challenge JSON from the cookies. Renamed the variable 'challengeStatus' to 'challengeDetails'

Comment (Mustafa-Esoofally) [reply src/app/api/challenge/start/route.ts:8]: Yes, when a start challenge request is sent we fetch the challenge from cookie. Then we check if it's a valid challenge and has duration and also if it's not started.

If all 3 conditions are satisfied we actually start the challenge

**Comment by Mustafa-Esoofally** [comment]
Made the changes as requested. Also, fixed a couple of build errors

File: src/app/admin/CandidateResponses.tsx (hunk 1 above)
Comment (Mustafa-Esoofally) [reply src/app/admin/CandidateResponses.tsx]: Fixed it

File: src/app/api/challenge/submit/route.ts (hunk 3 above)
Comment (Mustafa-Esoofally) [reply src/app/api/challenge/submit/route.ts]: Yeah, made the change

**Pull request merged**